<launch>

	<arg name="robot_name" default="robot1"/>
	<arg name="base_frame_id"   default="$(arg robot_name)/base_footprint"/>
  	<arg name="global_frame_id" default="$(arg robot_name)/map"/>
	<arg name="robot_name2" default="robot2"/>
	<arg name="base_frame_id2"   default="$(arg robot_name2)/base_footprint"/>
  	<arg name="global_frame_id2" default="$(arg robot_name2)/map"/>

	<arg name="topic_ns" default="costmap_to_voronoi"/>

	<group ns="$(arg robot_name)">
		<node pkg="costmap_to_voronoi" type="costmap_to_voronoi" name="costmap_to_voronoi" output="screen">
			<rosparam file="$(find costmap_to_voronoi)/params/costmap_common_params.yaml" command="load" ns="$(arg topic_ns)"/>
			<rosparam file="$(find costmap_to_voronoi)/params/global_costmap_params.yaml" command="load" ns="$(arg topic_ns)"/>
			<param name="$(arg topic_ns)/global_frame" value="$(arg global_frame_id)"/>
    		<param name="$(arg topic_ns)/robot_base_frame" value="$(arg base_frame_id)"/>
			<param name="topic_ns" value="$(arg topic_ns)"/>
			<remap from="/robot1/final_target" to="final_target"/>
		</node>
	</group>
	<group ns="$(arg robot_name2)">
		<node pkg="costmap_to_voronoi" type="costmap_to_voronoi" name="costmap_to_voronoi" output="screen">
			<rosparam file="$(find costmap_to_voronoi)/params/costmap_common_params.yaml" command="load" ns="$(arg topic_ns)"/>
			<rosparam file="$(find costmap_to_voronoi)/params/global_costmap_params.yaml" command="load" ns="$(arg topic_ns)"/>
			<param name="$(arg topic_ns)/global_frame" value="$(arg global_frame_id2)"/>
    		<param name="$(arg topic_ns)/robot_base_frame" value="$(arg base_frame_id2)"/>
			<param name="topic_ns" value="$(arg topic_ns)"/>
			<remap from="/robot2/final_target" to="final_target"/>
		</node>
	</group>
</launch>
